### ISSUE TITLE
Sending and receiving $type in payload

### ISSUE THREAD

**Issue by Tridy** [countincognito/Company.Dapr#1]
What are the options for not sending the **$type** in the parameter to the controller

`Company.Microservice.Membership/Membership/Register`

for example to send a request one needs to include the **$type** as a part of json body:

```
{
  "$type": "Company.Microservice.Membership.Data.v1_0.Web.RegisterRequestDto",
  "name": "My Name",
  "email": "whatever@gmail.com",
  "dateOfBirth": "2024-01-18T13:30:08.232Z"
}
```

especially since sending only the class name does not work, it needs the namespaces as well.

and it returns **$type** as a part of the body as well:

```
{
  "$type": "Company.Microservice.Membership.Data.v1_0.Web.RegisterResponseDto",
  "webMessage": "2024-01-18 13:30:08Z",
  "name": "My Name"
}
```

it is probably possible to do something based on the attribute or a property name, or something similar, but I was wondering if there are smarter ways of doing that and I would not be reinventing the wheel.

**Comment by countincognito** [comment]
You could just use concrete types for request and response objects (i.e. without polymorphic endpoints). It would be a standard API at that point.

The only reason I included typed requests/responses here was to demonstrate that polymorphism was possible over a REST/gRPC interface.

**Comment by Tridy** [comment]
I do want to have the polymorphism. It is that the convention is to use $type property to specify the full name, which could produce quite a long statement, for instance, `"$type": "Company.Microservice.Membership.Data.v1_0.Web.RegisterResponseDto"`.

I tried to see if there could be other ways to get the type from json by modifying the incoming json string, and reading the properties, but realized that sending json and not specifying the type potentially brings more trouble, especially when there are several properties from several DTOs mixed in one json. So, I shortened the namespace just for the DTOs, so `"$type": "Data.v1_0.Web.RegisterResponseDto"` looks a bit better now.

thanks for your work!
